textattack/metrics/quality_metrics/perplexity.py

```diff
@@ -0,0 +1,87 @@
+import torch
+from transformers import GPT2LMHeadModel, GPT2Tokenizer
+
+from textattack.attack_results import FailedAttackResult, SkippedAttackResult
+from textattack.metrics import Metric
+import textattack.shared.utils
+
+
+class Perplexity(Metric):
+    """Calculates average Perplexity on all successfull attacks using a pre-trained small GPT-2 model
+
+    Args:
+    results (:obj::`list`:class:`~textattack.goal_function_results.GoalFunctionResult`):
+                    Attack results for each instance in dataset
+    """
+
+    def __init__(self):
+        self.all_metrics = {}
+        self.original_candidates = []
+        self.successful_candidates = []
+        self.ppl_model = GPT2LMHeadModel.from_pretrained("gpt2")
+        self.ppl_model.to(textattack.shared.utils.device)
+        self.ppl_tokenizer = GPT2Tokenizer.from_pretrained("gpt2")
+        self.ppl_model.eval()
+        self.max_length = self.ppl_model.config.n_positions
+        self.stride = 128
```

Comment:
I think we can make it default longer than 128 (maybe like 512)?

Reply:
512